homeassistant/components/alert.py

```diff
@@ -0,0 +1,283 @@
+"""
+Support for repeating alerts when conditions are met.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/alert/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+import os
+
+import voluptuous as vol
+
+from homeassistant.core import callback
+from homeassistant.config import load_yaml_config_file
+from homeassistant.const import (CONF_ENTITY_ID, STATE_IDLE, CONF_NAME,
+                                 CONF_STATE, STATE_ON, STATE_OFF,
+                                 SERVICE_TURN_ON, SERVICE_TURN_OFF,
+                                 SERVICE_TOGGLE, ATTR_ENTITY_ID)
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.helpers import service, event
+from homeassistant.util import slugify
+from homeassistant.util.async import run_callback_threadsafe
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'alert'
+ENTITY_ID_FORMAT = DOMAIN + '.{}'
+
+CONF_BACKOFF = 'backoff'
+CONF_CAN_ACK = 'can_acknowledge'
+CONF_NOTIFIERS = 'notifiers'
+CONF_REPEAT = 'repeat'
+CONF_REPEAT_MAX = 'max_repeat'
+CONF_REPEAT_MIN = 'min_repeat'
+CONF_SKIP_FIRST = 'skip_first'
+
+ALERT_SCHEMA = vol.Schema({
+    # pylint: disable=no-value-for-parameter
+    vol.Required(CONF_NAME): cv.string,
+    vol.Required(CONF_ENTITY_ID): cv.string,
+    vol.Required(CONF_STATE, default=STATE_ON): cv.string,
+    vol.Required(CONF_REPEAT): cv.positive_int,
+    vol.Required(CONF_REPEAT_MAX, default=1440): cv.positive_int,
+    vol.Required(CONF_REPEAT_MIN, default=1): cv.positive_int,
+    vol.Required(CONF_BACKOFF, default=1): vol.Any(int, float),
+    vol.Required(CONF_CAN_ACK, default=True): cv.boolean,
+    vol.Required(CONF_SKIP_FIRST, default=False): cv.boolean,
+    vol.Required(CONF_NOTIFIERS): cv.ensure_list})
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        cv.string: ALERT_SCHEMA,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+
+ALERT_SERVICE_SCHEMA = vol.Schema({
+    vol.Required(ATTR_ENTITY_ID): cv.entity_ids,
+})
+
+
+def is_on(hass, entity_id):
+    """Return if the alert is firing and not acknowledged."""
+    return hass.states.is_state(entity_id, STATE_ON)
+
+
+def turn_on(hass, entity_id):
+    """Reset the alert."""
+    run_callback_threadsafe(hass.loop, async_turn_on, hass, entity_id)
```

Comment:
Ha, I literally requested the opposite. We want to implement each logic only once and thus call the async version of the same method.